.eslintrc.js

```diff
@@ -0,0 +1,48 @@
+/* eslint-env node */
+
+const PASCAL_CASE = "*([A-Z]*([a-z0-9]))";
+const CAMEL_CASE = "+([a-z])*([a-z0-9])*([A-Z]*([a-z0-9]))";
+
+module.exports = {
+  extends: [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
+    "plugin:react/recommended",
+    "plugin:react/jsx-runtime",
+    "plugin:prettier/recommended",
+  ],
+  env: {
+    browser: true,
+  },
+  parser: "@typescript-eslint/parser",
+  plugins: ["@typescript-eslint", "check-file"],
+  root: true,
+  settings: {
+    "import/resolver": {
+      typescript: true,
+      node: true,
+    },
+    react: {
+      version: "detect",
+    },
+  },
+  ignorePatterns: ["assets/**"],
+  rules: {
+    "@typescript-eslint/consistent-type-imports": "error",
+    "check-file/folder-naming-convention": [
+      "error",
+      { "src/**/": "KEBAB_CASE" },
+    ],
+    "check-file/filename-naming-convention": [
+      "error",
+      {
+        "src/**/*.{js,ts,tsx}": `@(${CAMEL_CASE}|${PASCAL_CASE})`,
```

Comment:
Unfortunately this is the only way to support multiple pattern per file name: https://github.com/dukeluo/eslint-plugin-check-file/issues/28#issuecomment-1643750137

I converted the file to `js` so we could at least name the variables but we can always inline the regex and keep it `.eslintrc.json`